sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -239,6 +239,16 @@ def delete_expired_environments(self) -> t.List[Environment]:
             self.environments_table,
             where=filter_expr,
         )
+        schema_names: t.Set[str] = set()
+        for environment in environments:
+            schema_names.update(
+                {
+                    exp.to_table(snapshot.qualified_view_name.for_environment(environment.name)).db
+                    for snapshot in environment.snapshots
+                }
+            )
+        for schema_name in schema_names:
+            self.engine_adapter.drop_schema(schema_name, ignore_if_not_exists=True, cascade=True)
```

Comment:
I don't think this logic belongs here. Our goal is to remove schemas in the DW, not in the state.

Reply:
This is where we delete individual views right now:
https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/core/context.py#L1167-L1168
Airflow: https://github.com/TobikoData/sqlmesh/blob/37d6a7fa3792a2fb641a279ca82c088aff4897cb/sqlmesh/engines/commands.py#L117-L118

Reply:
Man I keep getting this mixed up. I think it is because the SQLMesh installs I work with are storing state in the DW so I don't differentiate. Will need to take another look at this.